lib/logstash/inputs/snmp.rb

```diff
@@ -189,6 +220,15 @@ def validate_oids!
     raise(LogStash::ConfigurationError, "at least one get OID or one walk OID is required") if @get.empty? && @walk.empty?
   end
 
+  def validate_v3_user!
+    raise(LogStash::ConfigurationError, "v3 user must have a \"security_name\" option") if @security_name.nil?
+    raise(LogStash::ConfigurationError, "you must specify an auth protocol if you specify an auth pass") if  @auth_protocol.nil? && !@auth_pass.nil?
+    raise(LogStash::ConfigurationError, "you must specify an auth pass if you specify an auth protocol") if !@auth_protocol.nil? &&  @auth_pass.nil?
+    raise(LogStash::ConfigurationError, "you must specify a priv protocol if you specify a priv pass")   if  @priv_protocol.nil? && !@priv_pass.nil?
+    raise(LogStash::ConfigurationError, "you must specify a priv pass if you specify a priv protocol")   if !@priv_protocol.nil? &&  @priv_pass.nil?
+   end
```

Comment:
nitpick (resolve if you disagree): 
As I make mistakes and typos a lot I would like to see all my mistakes in one error message so I can fix it once and not fix the error presented to me and then see another one.
Will you consider pushing the error string to an array when the validation condition fails and raise a joined message once when the array is not empty?

Reply:
good suggestion!